docs/reference/builder.md

```diff
@@ -1172,8 +1172,10 @@ To use these, simply pass them on the command line using the `--build-arg
 `ARG` variables are not persisted into the built image as `ENV` variables are.
 However, `ARG` variables do impact the build cache in similar ways. If a
 Dockerfile defines an `ARG` variable whose value is different from a previous
-build, then a "cache miss" occurs upon its first usage, not its declaration.
-For example, consider this Dockerfile:
+build, then a "cache miss" occurs upon the first time it is possible to be used,
```

Comment:
I don't really like this sentence. It reads very awkwardly to me. What do you think of ..."occurs upon first use of the `ARG` variable. The declaration of the `ARG` variable does not count as a use." ? 

Perhaps some variation of the above.